Add error styles for `linter` and `spell-check`

jgierer12 opened the pull request:
Changes the styles for Atom's `linter` and `spell-check` packages to
match Nebula's existing error styles.

Fixes tjkohli/nebula-ui#6

jgierer12 committed:
Add error styles for `linter` and `spell-check`

Changes the styles for Atom's `linter` and `spell-check` packages to
match Nebula's existing error styles.

Fixes tjkohli/nebula-ui#6

jgierer12 committed:
Fix indentation

jgierer12 commented:
Thinking about it, it may be better to remove the pulse animation from the warning and info messages to make them a bit less intrusive.

tjkohli commented:
Thanks for your work!

I don't use the linter or spell-check packages, so can you please post some before/after screenshots here if possible? It'll help me document as well as understand the full effect of this change.

jgierer12 commented:
Before: ![Before](https://cloud.githubusercontent.com/assets/4331946/22414146/9a6d8ae0-e6bd-11e6-8433-7a015344a9e3.png)

After: ![After](https://cloud.githubusercontent.com/assets/4331946/22414145/9a581854-e6bd-11e6-8e89-ed30bf3a0534.png) (With the same pulsing animation you already had for the `.syntax--invalid.syntax--illegal` class)

Same for `spell-check`.

Ok, I'll remove the animation for warning and info then.

jgierer12 committed:
Remove pulse animation for warning and info

Removes the pulsing animation for warning and info messages in order to
make them less intrusive.

jgierer12 committed:
Refactor error highlighting styles

Moving styles for error highlighting to a mixin to reduce repetition.

jgierer12 committed:
Use recommended Less mixin syntax

tjkohli reviewed (commented):
Nicely done.

Gert-dev commented:
Thanks for fixing this!

However, I'm wondering about one thing: I understand that the pulsing is a bit intrusive... but... isn't that kind of the point? Error messages should be fixed as soon as possible or should be resolved in some other way (e.g. by adding an incorrectly spelled word to the dictionary). In no situation should they remain or be left alone, otherwise one risks mentally ignoring errors. This, however, only applies to errors and the situation may be different for warnings and information messages.

tjkohli commented:
@Gert-dev Error pulsing should be intrusive. Warnings and information? Probably not.

jgierer12 commented:
Yup, that's exactly what I've done. Errors pulse, warnings and errors just display the underline without pulsing.

tjkohli commented:
@jgierer12 I like your changes.

Though I don't use the linter so I'm not sure how these changes actually feel on a day-to-day basis, I assume it'll be okay.

I did modify the styles slightly since your commit, but it was more so defining the:

```
@background-color-info
@background-color-success
@background-color-warning
@background-color-error
```

In both the syntax and ui themes, and adjusted the shadow and pulse timing.

Gert-dev commented:
My bad, it appears I read about it being removed, then saw the non-animated screenshots, but missed the sentence where @jgierer12 mentioned that it was readded.

Keep up the great work!

Gert-dev commented:
@jgierer12 Sorry to bother you again about this. I've been trying to get this to work, but I keep getting the old, default linter styling with the dashed lines.

I see that the new CSS is active via my inspector, but it does not seem to want to apply.

![1](https://cloud.githubusercontent.com/assets/24809863/22525348/1f7b8a10-e8c7-11e6-9e2e-90e2e8e703a3.png)

However, looking at the CSS, it appears the selectors are incorrect - i.e. not those that are specified in the LESS file. When the Shadow DOM was deprecated in Atom 1.13, they started automatically transforming deprecated selectors into new syntax selectors, perhaps it's incorrectly transforming these as well? `error` seems to have become `syntax--error`, `trace` became `syntax--trace`, etc.

![2](https://cloud.githubusercontent.com/assets/24809863/22525377/38b8c5ec-e8c7-11e6-87a8-fcbb89004743.png)

![3](https://cloud.githubusercontent.com/assets/24809863/22525381/3c02d6fc-e8c7-11e6-884b-34037eceb93b.png)

tjkohli commented:
@Gert-dev please discuss this in one place, issue #3.
